src/ApolloInMemoryCache.re

```diff
@@ -36,22 +36,16 @@ let createIntrospectionFragmentMatcher = (~data) =>
   introspectionFragmentMatcher({"introspectionQueryResultData": data});
 
 /* Instantiate a new cache object */
+[@bs.obj] external
+makeApolloInMemoryCacheParams : 
+(
+  ~dataIdFromObject: (Js.t({..}) => string)=?, 
+  ~fragmentMatcher: (Js.t({..}) => string)=?
+) => _ = "";
+
 let createInMemoryCache = (~dataIdFromObject=?, ~fragmentMatcher=?, ()) => {
-  /* Apollo Client, looks for key in Object. Doesn't check if value is null  */
-  switch (dataIdFromObject, fragmentMatcher) {
-  | (Some(dataIdFromObject), Some(fragmentMatcher)) => 
-    apolloInMemoryCache({
-      "dataIdFromObject": Js.Nullable.return(dataIdFromObject),
-      "fragmentMatcher": Js.Nullable.return(fragmentMatcher)
-    })
-    | (Some(dataIdFromObject), None) => 
-    apolloInMemoryCache({
-      "dataIdFromObject": Js.Nullable.return(dataIdFromObject)
-    })
-    | (None, Some(fragmentMatcher)) => 
-    apolloInMemoryCache({
-      "fragmentMatcher": Js.Nullable.return(fragmentMatcher)
-    })
-    | (None, None) => apolloInMemoryCache(Js.Obj.empty())
-  };
+    /* Apollo Client, looks for key in Object. Doesn't check if value is null  */
+  apolloInMemoryCache(
+     makeApolloInMemoryCacheParams(~dataIdFromObject=?, ~fragmentMatcher=?)
```

Comment:
I’m not sure this is right. Punning optional args should be in this form 


makeApolloInMemoryCacheParams(~dataIdFromObject?, ~fragmentMatcher?)

Reply:
Notice no =

Reply:
Thanks for pointing out!! https://github.com/apollographql/reason-apollo/commit/6cc713fb4f2601ee67976c94f0f91bd5b4ad33bc